Add 'See also' sections to regex function docstring

GianmarcoCuppari opened the pull request:
Fixes #56752
    
    Added "See also" sections to the following regex function docstrings:
    - `match`: links to eachmatch, occursin, findfirst
    - `eachmatch`: links to match, findall, count
    - `count`: links to eachmatch, occursin, findall
    - `startswith`: links to endswith, occursin, match
    - `endswith`: links to startswith, occursin, match

GianmarcoCuppari committed:
Add 'See also' sections to regex function docstring
- Add cross-references between match, eachmatch, count
- Add cross-references between startswith and endswith
- Link related functions like occursin, findfirst, findall
- Addresses #56752

LilithHafner commented on base/regex.jl:
````diff
@@ -334,6 +334,9 @@ See also [`occursin`](@ref) and [`endswith`](@ref).
 ```
+# See also
+[`endswith`](@ref), [`occursin`](@ref), [`match`](@ref)
+
````
Adding `match` is a good idea, though it just needs to be added to the existing see also section above.

Same for the `endswith` docstring below

LilithHafner reviewed (commented):
One minor change and then LGTM!

GianmarcoCuppari committed:
Update See also section in regex.jl docstrings as suggested

GianmarcoCuppari commented:
I hadn’t noticed that some “See also” entries were already present.  
I’ve now merged the existing and newly added entries for `startswith` and `endswith`, as suggested.  
Thanks for pointing it out!

GianmarcoCuppari committed:
Fix whitespace and finalize docstring format in regex.jl

GianmarcoCuppari committed:
Add trailing newline to regex.jl

GianmarcoCuppari committed:
Merge branch 'master' into add-regex-see-also

GianmarcoCuppari committed:
Remove duplicate function references in regex docstrings

GianmarcoCuppari committed:
Merge branch 'add-regex-see-also' of https://github.com/GianmarcoCuppari/julia into add-regex-see-also

GianmarcoCuppari commented:
I've added "See also" cross-references to the regex docstrings:

- Added match reference to startswith and endswith
- Added a new "See also" section to match with references to eachmatch, occursin, and findfirst

However, the precompile test keeps failing with several errors..I removed all duplicate references thinking that was the issue, but the test still fails. I'm not sure what's causing this.

I'm not sure how to fix this. Could you help me understand what's happening and what I should do?

LilithHafner committed:
Merge branch 'master' into add-regex-see-also

IanButterworth committed:
Merge branch 'master' into add-regex-see-also
